beacon-chain/sync/validate_beacon_blocks.go

```diff
@@ -221,15 +230,26 @@ func (s *Service) validateBeaconBlock(ctx context.Context, blk block.SignedBeaco
 		return errors.New("incorrect proposer index")
 	}
 
-	return validateBellatrixBeaconBlock(parentState, blk.Block())
+	if err = s.validateBellatrixBeaconBlock(ctx, parentState, blk.Block()); err != nil {
+		s.setBadBlock(ctx, blockRoot)
```

Comment:
we would set bad blocks for blocks with optimistic parents, so this edge case needs to be handled